[IBM Z] : Test `test_check_pods_status_after_node_failure` failing eventhough the replacement pods are in `Running` state

Test fails check status of pods in `Terminating` state even though replacement pods are in Running state. Is this an expected behavior?

Test: `tests/manage/z_cluster/nodes/test_check_pods_status_after_node_failure.py::TestCheckPodsAfterNodeFailure::test_check_pods_status_after_node_failure`

Error:
The pods being checked are in the Terminating state and replacement pods are already available.
```
16:10:38 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc get node  -o yaml
16:10:38 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc -n openshift-storage get Pod rook-ceph-mds-ocs-storagecluster-cephfilesystem-b-54459799sc5b9 -n openshift-storage -o yaml
16:10:39 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc get node  -o yaml
16:10:39 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc -n openshift-storage get Pod rook-ceph-mgr-a-5cf998b95c-s4w5h -n openshift-storage -o yaml
16:10:39 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc get node  -o yaml
16:10:39 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc -n openshift-storage get Pod rook-ceph-mon-a-5464cb8bc5-w2qmr -n openshift-storage -o yaml

----

>           raise NotFoundError(f"Node name not found for the pod {pod_obj.name}")
E           ocs_ci.ocs.exceptions.NotFoundError: Node name not found for the pod rook-ceph-mon-a-5464cb8bc5-w2qmr

```

Mustgather and full logs: https://drive.google.com/file/d/1z0jgGfWm9Eddku470TUPAbiNLbQfWoun/view?usp=share_link

--- a/ocs_ci/ocs/node.py
+++ b/ocs_ci/ocs/node.py
@@ -17,6 +17,7 @@ from ocs_ci.ocs.exceptions import (
     NotAllNodesCreated,
     CommandFailed,
     ResourceNotFoundError,
+    NotFoundError,
 )
 from ocs_ci.ocs.ocp import OCP
 from ocs_ci.ocs.resources.ocs import OCS
@@ -1414,7 +1415,7 @@ def get_node_pods(node_name, pods_to_search=None, raise_pod_not_found_error=Fals
             if pod.get_pod_node(p).name == node_name:
                 node_pods.append(p)
         # Check if the command failed because the pod not found
-        except CommandFailed as ex:
+        except (CommandFailed, NotFoundError) as ex:
             if "not found" not in str(ex):
                 raise ex
             # Check the 2 cases of pod not found error

